base/dates/io.jl

```diff
@@ -159,14 +159,19 @@ function parse(x::AbstractString,df::DateFormat)
         throw(ArgumentError("Delimiter mismatch. Couldn't find first delimiter, \"$(df.slots[1].transition)\", in date string"))
     end
     periods = Period[]
-    extra = []
+    extra = Any[]  # Supports custom slot types such as TimeZone
```

Comment:
This shouldn't actually change anything, right? (since `[]` defaults to `Any[]`)